flake.nix

```diff
@@ -0,0 +1,26 @@
+{
+  description = "rust-gpu";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixpkgs-unstable";
```

Comment:
I'm not sure what the proper solution is but I will note that this approach has failed in the past:

https://github.com/Rust-GPU/rust-gpu-shadertoys/pull/43#issuecomment-3199776153

Keeping `default.nix` (as `shell.nix`, which I suppose is more appropriate), does make this less drastic, and I can explicitly avoid touching the flake path until it's usable for myself on NixOS etc.

But it's hard to tell what value of flakes really is outside of e.g. CI.